src/Microsoft.AspNetCore.ApplicationInsights.HostingStartup/JavaScriptSnippetTagHelperComponent.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights.AspNetCore;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.ApplicationInsights.HostingStartup
+{
+    /// <summary>
+    /// The <see cref="TagHelperComponent"/> that injects the <see cref="JavaScriptSnippet"/> to the head tag.
+    /// </summary>
+    public class JavaScriptSnippetTagHelperComponent : TagHelperComponent
+    {
+        private string _javaScriptSnippet;
+
+        /// <summary>
+        /// Initializes the <see cref="JavaScriptSnippetTagHelperComponent"/>.
+        /// </summary>
+        /// <param name="javaScriptSnippet">The <see cref="JavaScriptSnippet"/> to inject in the head tag.</param>
+        public JavaScriptSnippetTagHelperComponent(JavaScriptSnippet javaScriptSnippet)
+        {
+            _javaScriptSnippet = javaScriptSnippet.FullScript;
+        }
+
+        /// <inheritdoc />
+        public override int Order => 1;
```

Comment:
1 isn't a great order in-case you have other components. Should always leave room for others to do things around your execution without diving into negatives 😄 .

Do 100.